qucs/qucs-help/main.cpp

```diff
@@ -46,74 +47,29 @@ tQucsSettings QucsSettings; // application settings
-
-  QFile file(QDir::homePath()+QDir::convertSeparators ("/.qucs/helprc"));
-  if(!file.open(QIODevice::ReadOnly))
-    result = false; // settings file doesn't exist
-  else {
-    QTextStream stream(&file);
-    QString Line, Setting;
-    while(!stream.atEnd()) {
-      Line = stream.readLine();
-      Setting = Line.section('=',0,0);
-      Line = Line.section('=',1,1);
-      if(Setting == "HelpWindow") {
-	QucsSettings.x  = Line.section(",",0,0).toInt();
-	QucsSettings.y  = Line.section(",",1,1).toInt();
-	QucsSettings.dx = Line.section(",",2,2).toInt();
-	QucsSettings.dy = Line.section(",",3,3).toInt();
-	break; }
-    }
-    file.close();
-  }
-
-  file.setFileName(QDir::homePath()+QDir::convertSeparators ("/.qucs/qucsrc"));
-  if(!file.open(QIODevice::ReadOnly))
-    result = true; // qucs settings not necessary
-  else {
-    QTextStream stream(&file);
-    QString Line, Setting;
-    while(!stream.atEnd()) {
-      Line = stream.readLine();
-      Setting = Line.section('=',0,0);
-      Line = Line.section('=',1,1).trimmed();
-      if(Setting == "Font")
-	QucsSettings.font.fromString(Line);
-      else if(Setting == "Language")
-	QucsSettings.Language = Line;
-    }
-    file.close();
-  }
-  return result;
+  QSettings settings("qucs","qucs");
+  settings.beginGroup("QucsHelp");
+  if(settings.contains("x"))QucsSettings.x=settings.value("x").toInt();
+  if(settings.contains("y"))QucsSettings.y=settings.value("y").toInt();
+  if(settings.contains("dx"))QucsSettings.dx=settings.value("dx").toInt();
+  if(settings.contains("dy"))QucsSettings.dy=settings.value("dy").toInt();
+  settings.endGroup();
+  if(settings.contains("font"))QucsSettings.font.fromString(settings.value("font").toString());
+  if(settings.contains("Language"))QucsSettings.Language=settings.value("Language").toString();
+  return true;
 }
 
 // #########################################################################
 // Saves the settings in the settings file.
 bool saveApplSettings(QucsHelp *qucs)
 {
-  if(qucs->x() == QucsSettings.x)
-    if(qucs->y() == QucsSettings.y)
-      if(qucs->width() == QucsSettings.dx)
-	if(qucs->height() == QucsSettings.dy)
-	  return true;   // nothing has changed
-
-
-  QFile file(QDir::homePath()+QDir::convertSeparators ("/.qucs/helprc"));
-  if(!file.open(QIODevice::WriteOnly)) {
-    QMessageBox::warning(0, QObject::tr("Warning"),
-			QObject::tr("Cannot save settings !"));
-    return false;
-  }
-
-  QString Line;
-  QTextStream stream(&file);
-
-  stream << "Settings file, Qucs Help System " PACKAGE_VERSION "\n"
-	 << "HelpWindow=" << qucs->x() << ',' << qucs->y() << ','
-	 << qucs->width() << ',' << qucs->height() << '\n';
-
-  file.close();
+  QSettings settings ("qucs","qucs");
+  settings.beginGroup("QucsHelp");
+  settings.setValue("x", qucs->x());
+  settings.setValue("y", qucs->y());
+  settings.setValue("dx", qucs->width());
+  settings.setValue("dy", qucs->height());
+  settings.endGroup();
   return true;
```

Comment:
looking also at the similar `saveSettings()` functions around I understood that the function returned true when the settings were successfully saved or if the already saved settings are the same as the current application settings, which from the user's point of view is the same, I mean, if the function returned true, the user will have found the same settings when the application was restarted.
This was maybe useful in the past, when the settings were "manually" saved to a local file and not using `QSettings`. In any case I have the impression that the return value of the various `saveSettings()` function is actually never checked.